c7n/resources/s3.py

```diff
@@ -2612,8 +2612,114 @@ def process_bucket(self, bucket):
                 raise e
 
 
+class KMSKeyResolverMixin(object):
+    """Builds a dictionary of region specific ARNs"""
+
+    arns = dict()
+
+    def resolve_keys(self, buckets):
+        if 'key' not in self.data:
+            return None
+
+        regions = {get_region(b) for b in buckets}
+        for r in regions:
+            try:
+                client = local_session(self.manager.session_factory).client('kms', region_name=r)
+                self.arns[r] = client.describe_key(
+                    KeyId=self.data.get('key')
+                ).get('KeyMetadata').get('Arn')
+            except ClientError as e:
+                self.log.error('Error validating ARNs for set-bucket-encryption: %s' % e)
```

Comment:
s/`error resolving kms arns for set-bucket-encryption` also want to put in the self.data['key'] into the error output.